components/widgets/BatteryWidget.qml

```
@@ -44,151 +43,109 @@ OverviewWidget {
-
-		function update() {
-			let evenTargets = []
-			let oddTargets = []
-			for (let i = 0; i < animatedBarsRepeater.count; ++i) {
-				const animationTarget = animatedBarsRepeater.itemAt(i)
-				if (!animationTarget) {
-					// Don't set animation targets until all bar delegates have been initialized
-					return
-				}
-				if (i % 2 == 0) {
-					evenTargets.push(animationTarget)
-				} else {
-					oddTargets.push(animationTarget)
+			Row {
+				id: chimneysRow
+
+				height: parent.height
+
+				Repeater {
+					id: chimneyRepeater
+
+					model: Theme.animation.overviewPage.widget.battery.chimneys
+
+					delegate: Item {
+						id: chimney // a "chimney" which the bubbles rise up within.
+
+						required property int index
+
+						width: animationClip.width / Theme.animation.overviewPage.widget.battery.chimneys
+						height: root.expandedHeight // always full height, the clip item will clip it.
+						y: -(height - animationClip.height)
+
+						Repeater {
+							model: Theme.animation.overviewPage.widget.battery.bubbles
+							delegate: Rectangle {
+								id: bubble
+								required property int index
+								y: chimney.height
+								width: Theme.geometry.overviewPage.widget.battery.bubble.width
+								height: width
+								color: Theme.color.overviewPage.widget.battery.bubble.background
+								radius: height/2
+								border.width: 1
+								border.color: Theme.color.overviewPage.widget.battery.bubble.border
+
+								YAnimator {
+									id: yanimator
+									target: bubble
+									from: chimney.height
+									to: 0
+									duration: Theme.animation.overviewPage.widget.battery.bubble.duration + 100*Math.random()*bubble.index
+									easing.type: Easing.InOutQuad
+									loops: Animation.Infinite
+									running: root._animationReady && delayedStartTimer.count >= bubble.index
+								}
+
+								XAnimator {
+									target: bubble
+									from: bubble.index % 6 === 0 ? chimney.width/2 - bubble.width
```

The Repeater model uses a themed value `Theme.animation.overviewPage.widget.battery.bubbles` rather than hardcoded values, is there a reason for the `from` and `to` bindings use hardcoded values instead?

---

Good catch, will fix

---

Ah, actually, they aren't related to the bubbles value, but it's just meant to make sure the bubbles have slightly different tracks.  I will change to 3 instead of 6, and add a comment.